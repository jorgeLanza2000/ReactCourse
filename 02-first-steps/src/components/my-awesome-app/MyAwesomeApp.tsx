import type { CSSProperties } from 'react';

//? Las constantes es mejor que esten afuera de la función del componente
//? esto es porque cuando cambia de estado se ejecuta el componente entero entonces uno se puede ahorrar esa ejecución
const firstName = 'Jorge';
const lastName = 'Lanza';
const favoriteGames = ['CSGO', 'Valorant'];
const isActive = true; //! los valores bool no generan nada en html de react
//! los obj no se pueden imprimir solo así
const address = {
  zipCode: 'ABC-123',
  country: 'Canada',
};

//? tambies se pueden crear las css properties afuera
const myStyles: CSSProperties = {
  backgroundColor: 'blue',
  color: 'white',
  borderRadius: '10px',
  padding: 10,
};

export function MyAwesomeApp() {
  return (
    <div data-testid='div-app'>
      <h3>{lastName}</h3>
      <h1 data-testid='first-name-title'>{firstName}</h1>
      <p style={myStyles} className='mi-clase-favorita'>
        {favoriteGames.join(', ')}
      </p>
      {/* <h1>{isActive}</h1> */}
      <h1>{isActive ? 'Activo' : 'No activo'}</h1>
      {/* <p>{address}</p> */}
      <p
        style={{
          backgroundColor: 'red',
          borderRadius: 10,
          padding: '1rem',
          color: 'white',
        }}
      >
        {JSON.stringify(address)}
      </p>
    </div>
  );
}
