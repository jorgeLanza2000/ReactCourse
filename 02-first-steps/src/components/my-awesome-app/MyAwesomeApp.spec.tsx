import { describe, expect, test } from 'vitest';
import { MyAwesomeApp } from './MyAwesomeApp';
import { render, screen } from '@testing-library/react';

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />); //! solo se actualiza cuando se ejecuta el render
    // screen.debug(); //! se actualiza si algo se actualiza en el componente, por ejemplo un botÃ³n

    const h1 = container.querySelector('h1');
    expect(h1?.innerHTML).toContain('Jorge');

    const h3 = container.querySelector('h3');
    expect(h3?.innerHTML).toBe('Lanza');
  });

  test('should render firstName and lastName whit screen', () => {
    render(<MyAwesomeApp />);

    const h1 = screen.getByTestId('first-name-title');
    expect(h1.innerHTML).toBe('Jorge');
  });

  test('should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot(); //! es muy rigido ya que evalua la estructura html
  });

  test('should match snapshot con screen', () => {
    render(<MyAwesomeApp />);
    expect(screen.getByTestId('div-app')).toMatchSnapshot();
  });
});
