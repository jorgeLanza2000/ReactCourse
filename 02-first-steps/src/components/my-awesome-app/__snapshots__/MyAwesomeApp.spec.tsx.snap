// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h3>
      Lanza
    </h3>
    <h1
      data-testid="first-name-title"
    >
      Jorge
    </h1>
    <p
      class="mi-clase-favorita"
      style="background-color: blue; color: white; border-radius: 10px; padding: 10px;"
    >
      CSGO, Valorant
    </p>
    <h1>
      Activo
    </h1>
    <p
      style="background-color: red; border-radius: 10px; padding: 1rem; color: white;"
    >
      {"zipCode":"ABC-123","country":"Canada"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > should match snapshot con screen 1`] = `
<div
  data-testid="div-app"
>
  <h3>
    Lanza
  </h3>
  <h1
    data-testid="first-name-title"
  >
    Jorge
  </h1>
  <p
    class="mi-clase-favorita"
    style="background-color: blue; color: white; border-radius: 10px; padding: 10px;"
  >
    CSGO, Valorant
  </p>
  <h1>
    Activo
  </h1>
  <p
    style="background-color: red; border-radius: 10px; padding: 1rem; color: white;"
  >
    {"zipCode":"ABC-123","country":"Canada"}
  </p>
</div>
`;
